### PR TITLE
Update FQL 10 deploy and remove queries

### PR DESCRIPTION
**Pull request by pnwpedro**
Depends on https://github.com/fauna-labs/serverless-fauna/pull/50

### Changes
* Deploy query now returns `[{type, name, action, preview, original, result}, ...]`
* Deploy query now supports preview option (this is not yet used)
* Remove query now returns `[[{type, name, action, preview}, ...], ...]`
* Remove query now supports preview option (this is not yet used)
* Remove query calls toArray prior to map, eliminating pagination
* Remove query performs a more strict check for deletion policy: `f.data?.deletion_policy == "destroy"`

### CODE REVIEW

File: __tests__/commands/FQL10Functions.test.js [hunk 1]
```diff
@@ -73,6 +73,10 @@ describe("FQL 10 Functions", () => {
     await cleanup();
   });
 
+  afterAll(async () => {
+    faunaClient.close();
+  });
```
Comment (macmv) [inline __tests__/commands/FQL10Functions.test.js:78]: what does this do? it looks resonable, I'm just curious if it was fixing a bug related to other changes, or if theres some other reason this is here.

File: fauna/v10/queries/deploy.js [hunk 2]
```diff
@@ -1,44 +1,68 @@
...
-      let f = mod.byName(p.name)
-      if (Object.entries(p).every(v => v[1] == f[v[0]])) {
-        { type: mod.toString(), name: p.name, result: "noop" }
-      } else {
-        if (p.data != f.data) {
-          let pNullData = Object.assign(p, { data: null })
-          f.update(pNullData)
+    let p = p { name, body, data, role, }
+    
+    if (Function.byName(p.name) != null) {
```
Comment (macmv) [inline fauna/v10/queries/deploy.js:17]: style: putting a != and an else together makes a double negative in my head, so I'd switch around the ordering of the if/else. super minor though, this is just my preference

File: fauna/v10/queries/deploy.js [hunk 3]
```diff
@@ -68,16 +92,12 @@ const createUpdate = (module, params) => {
...
-    }
-    return fql`${prev}, ${curr}`;
-  }, null);
+  // The wire protocol doesn't yet support passing an array of queries, so
+  // we can manually construct the string parts.
+  const stringParts = ["[", ...queries.slice(0, -1).map((_) => ","), "]"];
 
-  return fql`${result}]`;
+  // Now pass the string parts and the queries to fql(), the backing function for fql``
+  return fql(stringParts, ...queries);
```
Comment (macmv) [inline fauna/v10/queries/deploy.js:102]: the json parser likes this change :)

**Review by macmv — APPROVED** [review]
LGTM, I just have a couple questions/minor suggestions

File: __tests__/commands/FQL10Functions.test.js (hunk 1 above)
Comment (pnwpedro) [reply __tests__/commands/FQL10Functions.test.js:78]: It leaves jest with open handles, so jest doesn't terminate cleanly :/

File: fauna/v10/queries/deploy.js (hunk 2 above)
Comment (pnwpedro) [reply fauna/v10/queries/deploy.js:17]: nice, i like that. update coming

**Pull request merged**